addons/point_of_sale/static/src/app/screens/payment_screen/payment_screen.js

```diff
@@ -212,19 +212,37 @@ export class PaymentScreen extends Component {
         this.hardwareProxy.openCashbox();
     }
     async addTip() {
-        // click_tip
         const tip = this.currentOrder.getTip();
         const change = this.currentOrder.getChange();
         const value = tip === 0 && change > 0 ? change : tip;
-
-        this.dialog.add(NumberPopup, {
+        const newTip = await makeAwaitable(this.dialog, NumberPopup, {
             title: tip ? _t("Change Tip") : _t("Add Tip"),
             startingValue: this.env.utils.formatCurrency(value, false),
             formatDisplayedValue: (x) => `${this.pos.currency.symbol} ${x}`,
-            getPayload: async (num) => {
-                await this.pos.setTip(parseFloat(num ?? ""));
-            },
         });
+
+        if (newTip === undefined) {
+            return;
+        }
+
+        await this.pos.setTip(parseFloat(newTip ?? ""));
+        const pLine =
+            this.selectedPaymentLine &&
+            (!this.selectedPaymentLine.isElectronic() ||
+                this.selectedPaymentLine.getPaymentStatus() === "pending")
+                ? this.selectedPaymentLine
+                : false;
+
+        if (!pLine || newTip === tip) {
+            this.notification.add(
+                _t(
+                    "The tip has been added to the order. However,the selected payment line does not allow tips to be added. "
+                )
+            );
```

Comment:
you should maybe not put this if there is no payment line as it makes no sense

Reply:
or if there is no tip (tip of 0)

Reply:
if there is an old tip it will reset it so there is